pkg/scalers/metrics_api_scaler.go

```diff
@@ -33,39 +32,87 @@ type metricsAPIScaler struct {
 }
 
 type metricsAPIScalerMetadata struct {
-	targetValue           float64
-	activationTargetValue float64
-	url                   string
-	format                APIFormat
-	valueLocation         string
-	unsafeSsl             bool
+	TargetValue           float64   `keda:"name=targetValue,order=triggerMetadata,optional"`
+	ActivationTargetValue float64   `keda:"name=activationTargetValue,order=triggerMetadata,default=0"`
+	URL                   string    `keda:"name=url,order=triggerMetadata"`
+	Format                APIFormat `keda:"name=format,order=triggerMetadata,default=json,enum=prometheus;json;xml;yaml"`
+	ValueLocation         string    `keda:"name=valueLocation,order=triggerMetadata"`
+	UnsafeSsl             bool      `keda:"name=unsafeSsl,order=triggerMetadata,default=false"`
+
+	// Auth method configuration
+	AuthMode string `keda:"name=authMode,order=triggerMetadata;authParams,optional"`
 
 	// apiKeyAuth
 	enableAPIKeyAuth bool
-	method           string // way of providing auth key, either "header" (default) or "query"
+	Method           string `keda:"name=method,order=triggerMetadata;authParams,default=header,enum=header;query"`
 	// keyParamName  is either header key or query param used for passing apikey
 	// default header is "X-API-KEY", default query param is "api_key"
-	keyParamName string
-	apiKey       string
+	KeyParamName string `keda:"name=keyParamName,order=triggerMetadata;authParams,optional"`
+	APIKey       string `keda:"name=apiKey,order=authParams,optional"`
 
 	// base auth
 	enableBaseAuth bool
-	username       string
-	password       string // +optional
+	Username       string `keda:"name=username,order=authParams,optional"`
+	Password       string `keda:"name=password,order=authParams,optional"` // +optional
 
 	// client certification
 	enableTLS bool
-	cert      string
-	key       string
-	ca        string
+	Cert      string `keda:"name=cert,order=authParams,optional"`
+	Key       string `keda:"name=key,order=authParams,optional"`
+	CA        string `keda:"name=ca,order=authParams,optional"`
 
 	// bearer
 	enableBearerAuth bool
-	bearerToken      string
+	BearerToken      string `keda:"name=token,order=authParams,optional"`
```

Comment:
curious if this can be further simplified by using `authentication.Config` like in https://github.com/kedacore/keda/pull/6969? So far each scaler had its own quirky behavior and expectations regarding auth params so if it's too problematic, we can merge as is and work on further improvements later